impl/src/main/java/org/ehcache/config/CacheConfigurationBuilder.java

```diff
@@ -239,6 +240,24 @@ public boolean hasDefaultExpiry() {
     return otherBuilder;
   }
 
+  public CacheConfigurationBuilder<K, V> withSizeOfEngine(long maxDepth, long maxSize) {
+    CacheConfigurationBuilder<K, V> otherBuilder = new CacheConfigurationBuilder<K, V>(this);
+    otherBuilder.add(new DefaultSizeOfEngineConfiguration(maxDepth, maxSize));
+    return otherBuilder;
+  }
+
+  public CacheConfigurationBuilder<K, V> withSizeOfEngineOfMaxDepth(long maxDepth) {
+    CacheConfigurationBuilder<K, V> otherBuilder = new CacheConfigurationBuilder<K, V>(this);
+    otherBuilder.add(new DefaultSizeOfEngineConfiguration(maxDepth, Integer.MAX_VALUE));
```

Comment:
You cannot keep adding configurations like that. You need to make sure in case there is one, to _copy_ it and adapt it.